Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -0,0 +1,677 @@
+            return {1,1};
+        }
+        
+        void DrawFrame() {
+            @autoreleasepool {
+                // Create a new command buffer for each render pass to the current drawable.
+                id<MTLCommandBuffer> commandBuffer = [commandQueue commandBuffer];
+                commandBuffer.label = @"XRDisplayCommandBuffer";
+                
+                id<CAMetalDrawable> drawable = [MetalLayer nextDrawable];
+                MTLRenderPassDescriptor *renderPassDescriptor = [MTLRenderPassDescriptor renderPassDescriptor];
+
+                if(renderPassDescriptor != nil) {
+                    renderPassDescriptor.colorAttachments[0].texture = drawable.texture;
+                    renderPassDescriptor.colorAttachments[0].loadAction = MTLLoadActionClear;
+                    renderPassDescriptor.colorAttachments[0].clearColor = MTLClearColorMake(0.0,0.0,0.0,1.0);
+                    
+                    // Create a render command encoder.
+                    id<MTLRenderCommandEncoder> renderEncoder = [commandBuffer renderCommandEncoderWithDescriptor:renderPassDescriptor];
+                    renderEncoder.label = @"XRDisplayEncoder";
+
+                    // Set the region of the drawable to draw into.
+                    [renderEncoder setViewport:(MTLViewport){0.0, 0.0, static_cast<double>(viewportSize.x), static_cast<double>(viewportSize.y), 0.0, 1.0 }];
+                    
+                    [renderEncoder setRenderPipelineState:pipelineState];
+
+                    // Pass in the parameter data.
+                    [renderEncoder setVertexBytes:vertices length:sizeof(vertices) atIndex:0];
+
+                    [renderEncoder setFragmentTexture:id<MTLTexture>(ActiveFrameViews[0].ColorTexturePointer) atIndex:0];
+                    [renderEncoder setFragmentTexture:sessionDelegate.cameraTextureY atIndex:1];
+                    [renderEncoder setFragmentTexture:sessionDelegate.cameraTextureCbCr atIndex:2];
+
+                    // Draw the triangles.
+                    [renderEncoder drawPrimitives:MTLPrimitiveTypeTriangleStrip vertexStart:0 vertexCount:4];
+
+                    [renderEncoder endEncoding];
+
+                    // Schedule a present once the framebuffer is complete using the current drawable.
+                    [commandBuffer presentDrawable:drawable];
+                }
+
+                // Finalize rendering here & push the command buffer to the GPU.
+                [commandBuffer commit];
+            }
+        }
+
+        void GetHitTestResults(std::vector<HitResult>&, xr::Ray) const {
+            // TODO
+        }
+        
+        private:
+            inline static ARSession* session{};
+            inline static ARWorldTrackingConfiguration* configuration{};
+            MTKView* xrView{};
+            bool SessionEnded{ false };
+            id<MTLDevice> MetalDevice{};
+            CAMetalLayer* MetalLayer{};
```

Comment:
Seems like there is a mix of casing here. Should these three be `sessionEnded`, `metalDevice`, `metalLayer`?

Reply:
Fixed